src/csharp/Grpc.Core/Internal/AuthContextSafeHandle.cs

```diff
@@ -74,7 +74,7 @@ protected override bool ReleaseHandle()
 
         private AuthProperty PtrToAuthProperty(IntPtr authPropertyPtr)
         {
-            var nativeAuthProperty = (NativeAuthProperty) Marshal.PtrToStructure(authPropertyPtr, typeof(NativeAuthProperty));
+            var nativeAuthProperty = Marshal.PtrToStructure<NativeAuthProperty>(authPropertyPtr);
```

Comment:
there are a couple of test failures with:


```
2017-08-09 07:57:11,100 ++ dirname tools/run_tests/helper_scripts/build_csharp.sh
+ cd tools/run_tests/helper_scripts/../../../src/csharp
+ '[' dbg == gcov ']'
+ dotnet build --configuration Debug Grpc.sln
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

Internal/AuthContextSafeHandle.cs(77,46): error CS0308: The non-generic method 'Marshal.PtrToStructure(IntPtr, object)' cannot be used with type arguments [/var/local/git/grpc/src/csharp/Grpc.Core/Grpc.Core.csproj]

Build FAILED.

Internal/AuthContextSafeHandle.cs(77,46): error CS0308: The non-generic method 'Marshal.PtrToStructure(IntPtr, object)' cannot be used with type arguments [/var/local/git/grpc/src/csharp/Grpc.Core/Grpc.Core.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:14.31

2017-08-09 07:57:11,100 FAILED: tools/run_tests/helper_scripts/build_csharp.sh [ret=1, pid=6102]
2017-08-09 07:57:11,100 FAILED: Some tests failed
+ EXITCODE=1
+ docker rm -f build_grpc_interop_csharp_94ce22d8-1837-4750-b375-e1a8b9f5374b
build_grpc_interop_csharp_94ce22d8-1837-4750-b375-e1a8b9f5374b
+ exit 1

2017-08-09 07:57:15,196 FAILED: build_docker_csharp [ret=1, pid=25582]
```

Not sure exactly why, but did see that `Marshal.PtrToStructure<NativeAuthProperty>` is introduced in net45.1